pkgs/applications/misc/blender/default.nix

```diff
@@ -1,142 +1,136 @@
-{ config, stdenv, lib, fetchurl, fetchzip, boost, cmake, ffmpeg, gettext, glew
-, ilmbase, libXi, libX11, libXext, libXrender
-, libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio2, openjpeg, python39Packages
-, openvdb, libXxf86vm, tbb, alembic
-, zlib, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
-, jackaudioSupport ? false, libjack2
-, cudaSupport ? config.cudaSupport or false, cudatoolkit
-, colladaSupport ? true, opencollada
-, spaceNavSupport ? stdenv.isLinux, libspnav
-, makeWrapper
-, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
-, potrace
-, openxr-loader
-, embree, gmp, libharu
+{ stdenv, lib, config, fetchzip, cmake, makeWrapper
+# Common:
+, alembic, boost, embree, ffmpeg, fftw, freetype, gettext
+, glew, gmp, ilmbase, jemalloc, libharu, libjpeg, libpng
+, libsamplerate, libsndfile, libtiff, opencolorio, openexr
+, openimagedenoise, openimageio2, openjpeg, opensubdiv
+, potrace, pugixml, python3, tbb, zlib
+
+# Linux specific:
+, libGL, libGLU, libX11, libXext, libXi, libXrender, libXxf86vm
+, openxr-loader, openvdb, openal, ocl-icd
+
+# Darwin specific:
+, Cocoa, CoreGraphics, ForceFeedback, llvmPackages, OpenAL, OpenGL, SDL
+
+# Optional:
+, jackaudioSupport ? false
+, libjack2
+
+, cudaSupport ? config.cudaSupport or false
+, cudatoolkit, addOpenGLRunpath
+
+, colladaSupport ? true
+, opencollada
+
+, spaceNavSupport ? stdenv.isLinux
+, libspnav
 }:
 
-with lib;
 let
-  python = python39Packages.python;
   optix = fetchzip {
+    # As a reference check the URL against this package in the ArchLinux repo
+    # and against what OpenShadingLanguage has in its github workflows
     url = "https://developer.download.nvidia.com/redist/optix/v7.0/OptiX-7.0.0-include.zip";
     sha256 = "1b3ccd3197anya2bj3psxdrvrpfgiwva5zfv2xmyrl73nb2dvfr7";
   };
 
+  numpyPath = "${python3.pkgs.numpy}/${python3.sitePackages}";
 in
 stdenv.mkDerivation rec {
   pname = "blender";
   version = "2.93.5";
 
-  src = fetchurl {
+  src = fetchzip {
```

Comment:
The benefit of using `fetchurl` is that it allows for tarballs to be mirrored on tarballs.nixos.org

Reply:
But fetchzip is a wrapper around fetchurl, how does that work?

Reply:
https://github.com/NixOS/nixpkgs/blob/2f218ac01d9d1c1ef8cca1cd0eb13d4971209b89/maintainers/scripts/find-tarballs.nix#L19-L23

`drv.outputHashMode == "flat"` is required for some reason